### PR TITLE
Add Package entity

### PR DESCRIPTION
**Pull request by ehelms**

### CODE REVIEW

File: nailgun/entities.py [hunk 1]
```diff
@@ -3811,6 +3811,32 @@ def list_smart_variables(self, synchronous=True, **kwargs):
         return _handle_response(response, self._server_config, synchronous)
 
 
+class Package(Entity, EntityReadMixin, EntitySearchMixin):
+    """A representation of a Package entity."""
+
+    def __init__(self, server_config=None, **kwargs):
+        self._fields = {
+            'arch': entity_fields.StringField(),
+            'checksum': entity_fields.StringField(),
```
Comment (omaciel) [inline nailgun/entities.py]: Should we add the valid choices for this field?

File: nailgun/entities.py [hunk 2]
```diff
@@ -3811,6 +3811,32 @@ def list_smart_variables(self, synchronous=True, **kwargs):
...
+            'description': entity_fields.StringField(),
+            'epoch': entity_fields.StringField(),
+            'filename': entity_fields.StringField(),
+            'name': entity_fields.StringField(
```
Comment (omaciel) [inline nailgun/entities.py]: So only `name` is a required field?

File: nailgun/entities.py [hunk 3]
```diff
@@ -3811,6 +3811,32 @@ def list_smart_variables(self, synchronous=True, **kwargs):
...
+                str_type='alpha',
+                length=(6, 12)
```
Comment (svtkachenko) [inline nailgun/entities.py]: I think `lenth` is used when creating entity, for read-only entities there is no use in this.

File: nailgun/entities.py (hunk 1 above)
Comment (ehelms) [inline nailgun/entities.py]: What are the valid choices? Checksums are giant hex strings.

File: nailgun/entities.py (hunk 2 above)
Comment (ehelms) [inline nailgun/entities.py]: Actually, I don't think you can lookup by name only search by it, so I guess I should remove this? I added fields based on what the API returns for a package vs. the parameters you can pass TO the API. Which should this set of attributes be?

Comment (omaciel) [inline nailgun/entities.py]: `name` would be populated for you automatically on `create`, so I was wondering what are the minimal attributes one must specify in order to create a package?

**Comment by ehelms** [comment]
One does not simply create a package. They are read only, since they come
from sync or upload to a repo.

**Pull request merged**